sympy/core/basic.py

```diff
@@ -1525,6 +1525,39 @@ def match(self, pattern, old=False):
         pattern = sympify(pattern)
         return pattern.matches(self, old=old)
 
+    def match_between(self, pattern1, pattern2, old=False):
+        """
+        match_between compares the given two patterns, and suggests the one
+        with more variables matched. This will help less initialization of
+        variables which will be helpful in Rubi. Example:
```

Comment:
Can you explain how having fewer variables being assigned is preferred? It would seem to me that you would just want to return the first expression that matches (you could sort the patterns in such a way that the more preferable one is checked first), so that you don't do unnecessary matching for performance.